crates/core_simd/src/swizzle.rs

```diff
@@ -364,4 +364,131 @@ where
 
         (Even::swizzle2(self, other), Odd::swizzle2(self, other))
     }
+
+    /// Takes a slice of a vector to produce a shorter vector.
+    ///
+    /// This is equivalent to computing `&self[OFFSET..OFFSET+LEN]` on
+    /// the underlying array.
+    ///
+    /// ```
+    /// # #![feature(portable_simd)]
+    /// # use core::simd::Simd;
+    /// let x = Simd::from_array([0, 1, 2, 3, 4, 5, 6, 7]);
+    /// let y = x.slice::<2, 4>();
+    /// assert_eq!(y.to_array(), [2, 3, 4, 5]);
```

Comment:
The example is good on its own, but I'd like to know: what sort of algorithms are you going to use this one in? I'd like to see this one in action. For the benefit of deciding what this should look like in the future, not necessarily for inclusion in the docs.